rules/share-your-developer-secrets-securely/rule.md

```diff
@@ -0,0 +1,112 @@
+---
+type: rule
+archivedreason:
+title: Do you share your developer secrets securely?
+guid: 6165deaa-a3b0-40b1-8659-482a34faf7c4
+uri: share-your-developer-secrets-securely
+created: 2023-08-14T00:00:00.0000000Z
+authors:
+  - title: Gordon Beeming
+    url: https://ssw.com.au/people/gordon-beeming
+related: 
+  - store-your-secrets-securely
+redirects: []
+---
+
+Most systems will have variables that need to be stored securely; OpenId shared secret keys, connection strings, and API tokens to name a few.
+
+These secrets **must not** be stored in source control. It is insecure and means they are sitting out in the open, wherever code has been downloaded, for anyone to see.
+
+[Do you store your secrets securely?](https://www.ssw.com.au/rules/store-your-secrets-securely/) shows different ways to store your secrets securely. When you use .NET User Secrets, you can store your secrets in a JSON file on your local machine. This is great for development, but how do you share those secrets securely with other developers in your organization?
+
+<!--endintro-->
+
+### Bad Examples
+
+::: greybox
+
+#### Do not store secrets in appsettings.Development.json
+
+The `appsettings.Development.json` file is meant for storing development settings. It is not meant for storing secrets. This is a bad practice because it means that the secrets are stored in source control, which is insecure.
+
+![](development-json.jpg)
+
+:::
+::: bad
+Figure: Bad practice - Overall rating: 1/10
```

Comment:
0/10? 😜

Reply:
I wasn't sure it went that low 😂

Reply:
I can update, used 1 to match the other rule. happy to change both. I guess 0 would be for public code where 1 is if the source control system itself is at least private